client/src/components/RegisterAccountForm/RegisterAccountForm.js

```diff
@@ -27,31 +30,37 @@ const RegisterAccountForm = () => {
   }
 
   const handleSubmit = event => {
-    /*
-    The alerts (and this comment) will be removed; they are just here for testing and giving some feedback
-    */
     event.preventDefault()
-    alert(
-      `SUBMIT WAS INITIATED\n${username}\n${email}\n${password}\n${passwordReType}`
-    )
     let isErrorInRegistration = false
     if (password !== passwordReType) {
       isErrorInRegistration = true
-      setErrorRegister('Passwords do not match!')
-      alert('Passwords do not match')
+      if (successMessage !== '') setSuccessMessage('')
+      setErrorMessage('Passwords do not match!')
     }
 
     if (!isErrorInRegistration) {
-      alert('SUCCESS!')
-      event.target.reset()
-      if (errorRegister !== '') setErrorRegister('')
+      axios
+        .post(`/api/users`, { username, email, password })
```

Comment:
You might want to put the axios request into its own function.

It might also be a good idea to create a file with a function for every request of the app and put them all together in one dedicated directory. This would allow you get an overview of every request the app makes to the outside world.